hibernate-reactive-core/src/main/java/org/hibernate/reactive/mutiny/impl/MutinySessionFactoryImpl.java

```diff
@@ -34,37 +36,47 @@
 public class MutinySessionFactoryImpl implements Mutiny.SessionFactory {
 
 	private SessionFactoryImpl delegate;
+	private final VertxInstance vertxInstance;
 
 	public MutinySessionFactoryImpl(SessionFactoryImpl delegate) {
 		this.delegate = delegate;
+		vertxInstance = delegate.getServiceRegistry().getService(VertxInstance.class);
+	}
+
+	<T> Uni<T> uni(Supplier<CompletionStage<T>> stageSupplier) {
+		return Uni.createFrom().completionStage(stageSupplier)
+				.runSubscriptionOn( runnable -> vertxInstance.getVertx()
+						.getOrCreateContext().runOnContext( x -> runnable.run() ) );
```

Comment:
So, @vietj @cescoffier @Sanne just how expensive is this?

Reply:
What we could do is something like:

```java
runnable -> {
   Context c = vertxInstance.getVertx().getOrCreateContext();
   if (Vertx.currentContext() == c) { 
      runnable.run();
   } else { 
      c.runOnContext(() -> runnable.run());
   }  

}
```

Reply:
I'd need to run some benchmarks but perhaps there's a better way out:

could we first check if we're on a context, then avoid this all if we already are?

For those users who are not running on the vert.x context, it would be more acceptable to have a performance penalty - they probably don't care as much.

Reply:
Well I would probably still want to make it lazy, in order to get session operation atomicity, but yes, we can definitely check if there's a context first.